batched/dense/unit_test/Test_Batched_SerialPbtrs.hpp

```diff
@@ -0,0 +1,295 @@
+    Kokkos::parallel_for(name.c_str(), policy, *this);
+  }
+};
+
+template <typename DeviceType, typename ScalarType, typename LayoutType, typename ParamTagType, typename AlgoTagType>
+/// \brief Implementation details of batched pbtrs test
+///        Confirm A * x = b, where
+///        A: [[4, 1, 0],
+///            [1, 4, 1],
+///            [0, 1, 4]]
+///        b: [1, 1, 1]
+///        x: [3/14, 1/7, 3/14]
+///
+///        This corresponds to the following system of equations:
+///        4 x0 +   x1        = 1
+///          x0 + 4 x1 +   x2 = 1
+///                 x1 + 4 x2 = 1
+/// \param N [in] Batch size of RHS (banded matrix can also be batched matrix)
+/// \param k [in] Number of superdiagonals or subdiagonals of matrix A
+/// \param BlkSize [in] Block size of matrix A
+void impl_test_batched_pbtrs_analytical(const int N) {
+  using ats        = typename Kokkos::ArithTraits<ScalarType>;
+  using RealType   = typename ats::mag_type;
+  using View2DType = Kokkos::View<ScalarType **, LayoutType, DeviceType>;
+  using View3DType = Kokkos::View<ScalarType ***, LayoutType, DeviceType>;
+
+  constexpr int BlkSize = 3, k = 1;
+  View3DType A("A", N, BlkSize, BlkSize), A_reconst("A_reconst", N, BlkSize, BlkSize);
+  View3DType Ab("Ab", N, k + 1, BlkSize);                                             // Banded matrix
+  View2DType x0("x0", N, BlkSize), x_ref("x_ref", N, BlkSize), y0("y0", N, BlkSize);  // Solutions
+
+  auto h_A_reconst = Kokkos::create_mirror_view(A_reconst);
+  auto h_x_ref     = Kokkos::create_mirror_view(x_ref);
+
+  for (int ib = 0; ib < N; ib++) {
+    for (int i = 0; i < BlkSize; i++) {
+      for (int j = 0; j < BlkSize; j++) {
+        h_A_reconst(ib, i, j) = i == j ? 4.0 : 1.0;
+      }
+    }
+
+    h_x_ref(ib, 0) = 3.0 / 14.0;
+    h_x_ref(ib, 1) = 1.0 / 7.0;
+    h_x_ref(ib, 2) = 3.0 / 14.0;
+  }
+
+  Kokkos::fence();
+
+  Kokkos::deep_copy(x0, ScalarType(1.0));
+  Kokkos::deep_copy(A_reconst, h_A_reconst);
+
+  // Create banded triangluar matrix in normal and banded storage
+  using ArgUplo = typename ParamTagType::uplo;
+  create_banded_pds_matrix<View3DType, View3DType, ArgUplo>(A_reconst, A, k, false);
+  create_banded_triangular_matrix<View3DType, View3DType, ArgUplo>(A_reconst, Ab, k, true);
+
+  // Factorize with Pbtrf: A = U**H * U or A = L * L**H
+  Functor_BatchedSerialPbtrf<DeviceType, View3DType, ParamTagType, AlgoTagType>(Ab).run();
+
+  // pbtrs (Note, Ab is a factorized matrix of A)
+  auto info = Functor_BatchedSerialPbtrs<DeviceType, View3DType, View2DType, ParamTagType, AlgoTagType>(Ab, x0).run();
+
+  Kokkos::fence();
+  EXPECT_EQ(info, 0);
+
+  // this eps is about 10^-14
+  RealType eps = 1.0e3 * ats::epsilon();
+  auto h_x0    = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), x0);
+
+  // Check x0 = x1
+  for (int ib = 0; ib < N; ib++) {
+    for (int i = 0; i < BlkSize; i++) {
+      EXPECT_NEAR_KK(h_x0(ib, i), h_x_ref(ib, i), eps);
```

Comment:
For this type of floating point error checking it makes sense to use `Test::EXPECT_NEAR_KK_REL` as long as the value tested is non-zero. The relative error checking is usually more precise and always you to chose epsilon accurately.

Reply:
Sure. Now using `Test::EXPECT_NEAR_KK_REL` instead